src/app/cli/src/cli_commands.rs

```diff
@@ -574,45 +488,34 @@ pub fn get_command(
                 cli_catalog.get_one()?,
                 cli_catalog.get_one()?,
                 cli_catalog.get_one()?,
-                submatches.get_one("address").copied(),
-                submatches.get_one("http-port").copied(),
-                submatches.get_flag("get-token"),
+                c.address,
+                c.http_port,
+                c.get_token,
             ))
         }
-        Some(("verify", submatches)) => Box::new(VerifyCommand::new(
+        cli::Command::Verify(c) => Box::new(VerifyCommand::new(
             cli_catalog.get_one()?,
             cli_catalog.get_one()?,
             cli_catalog.get_one()?,
             cli_catalog.get_one()?,
             cli_catalog.get_one()?,
-            validate_many_dataset_patterns(
-                cli_catalog,
-                submatches.get_many("dataset").unwrap().cloned(),
-            )?
-            .into_iter(),
-            submatches.get_flag("recursive"),
-            submatches.get_flag("integrity"),
-        )),
-        Some(("version", submatches)) => Box::new(VersionCommand::new(
-            cli_catalog.get_one()?,
-            submatches.get_one("output-format").map(String::as_str),
+            validate_many_dataset_patterns(cli_catalog, c.dataset)?.into_iter(),
+            c.recursive,
+            c.integrity,
         )),
-        _ => return Err(CommandInterpretationFailed.into()),
+
+        cli::Command::Version(c) => {
+            Box::new(VersionCommand::new(cli_catalog.get_one()?, c.output_format))
+        }
     };
 
     Ok(command)
 }
 
-pub fn command_needs_transaction(arg_matches: &clap::ArgMatches) -> Result<bool, CLIError> {
-    match arg_matches.subcommand() {
-        Some(("system", system_matches)) => match system_matches.subcommand() {
-            Some(("generate-token", _)) => Ok(true),
-            Some(_) => Ok(false),
-            None => Err(CommandInterpretationFailed.into()),
-        },
-        Some(("delete", _)) => Ok(true),
```

Comment:
We need a transaction for deletion: saving the Outbox messages requires a transaction, in case of running with databases. But I don't mind if you delete in this PR.

In https://github.com/kamu-data/kamu-cli/pull/787/, I add processing of the Outbox messages (as well as the transaction for `add`), I can also return back `delete`.

Reply:
Thanks for noticing - removing delete was not intentional. Fixed.